### PR TITLE
add return to core/header

### LINKED ISSUE #2994: Improve setupHeader function
### Describe the problem

setupHeader function only echos output and it should also support return of output.

### Describe the solution you'd like

Add a second parameter to the setupHeader function to instead return the value rather than echo it. so function in src/Core/Header.php could be something like:
```
public static function setupHeader($assets = [], $echoOutput = true)
```
Then when this is working (incorporate ability to return rather than echo it), adding a second parameter with false will then work. When have above mechanism working, a perfect place to try this is the setupHeader call in interface/patient_file/printed_fee_sheet.php

### PR DESCRIPTION
**Pull request by stephenwaite**
Fixes #2994 

#### Short description of what this resolves:
the now closed #2990 

#### Changes proposed in this pull request:

following at @bradymiller 's advice, sorry @thesmallstar but maybe this needs more work since haven't tested yet

### CODE REVIEW

File: interface/patient_file/printed_fee_sheet.php [hunk 1]
```diff
@@ -313,9 +313,9 @@ function genColumn($ix)
 }
 </style>";
 
-$html .= "<title>" . text($frow['name']) . "</title>
-".Header::setupHeader(['opener', 'topdialog', 'dialog'])."
-<script language=\"JavaScript\">";
+$html .= "<title>" . text($frow['name']) . "</title>" .
+    Header::setupHeader(['opener', 'topdialog', 'dialog'], $echoOutput = false) .
```
Comment (bradymiller) [inline interface/patient_file/printed_fee_sheet.php]: change 2nd parameter to just false:
```
Header::setupHeader(['opener', 'topdialog', 'dialog'], false) .
```

File: src/Core/Header.php [hunk 2]
```diff
@@ -28,6 +28,7 @@ class Header
     private static $scripts;
     private static $links;
     private static $isHeader;
+    private static $output;
```
Comment (bradymiller) [inline src/Core/Header.php]: drop above line

File: src/Core/Header.php [hunk 3]
```diff
@@ -72,11 +73,16 @@ class Header
...
      */
-    public static function setupHeader($assets = [])
+    public static function setupHeader($assets = [], $echoOutput = true)
     {
         // Required tag
-        echo '<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />';
-        self::setupAssets($assets, true);
+        $output = '<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no" />';
+        $output .= self::setupAssets($assets, true);
+        if ($echoOutput == true) {
```
Comment (bradymiller) [inline src/Core/Header.php]: could simplify above, but just doing:
```
if ($echoOutput) {
```

File: src/Core/Header.php [hunk 4]
```diff
@@ -72,11 +73,16 @@ class Header
...
+            echo $output;
+        } else {
+            return $output;
+        }
     }
```
Comment (bradymiller) [inline src/Core/Header.php]: Then just need to change the following line:
https://github.com/openemr/openemr/blob/081634938d424d99c83723fa2a59609c94aceee7/src/Core/Header.php#L105
to:
```
return self::includeAsset($assets);
```

**Comment by bradymiller** [comment]
hi @stephenwaite ,
With above changes, I think it will work. There are 2 possible outcomes:
1. 90% chance that it works flawlessly
2. 10% chance that it causes a white screen of death leading to a seismic disaster causing the Northeastern US to plunge into the frigid Atlantic (no pressure, though :) I'll keep an eye on the news to see how it goes :) ).

**Comment by thesmallstar** [comment]
@stephenwaite , thank you for picking this up. I will help you with the testing and replacing the the `Header::setupHeader` function on other places where it used (like issues popup).

**Comment by thesmallstar** [comment]
Tested, and it works perfectly!

**Comment by stephenwaite** [comment]
thank you for testing @thesmallstar , did notice that when clicked Print and then cancel the Print button disappeared, not a big deal but found it interesting

**Comment by thesmallstar** [comment]
Yes, I did notice that. 
The div has class `hideonprint` , and this seems to be doing that, also the class being so specific I thought that must be intentional. 
Is that needed though?

**Comment by stephenwaite** [comment]
hi @thesmallstar, thanks for the explanation, it's fine the way it is, so will ask Brady to take another look and hopefully bring this in.

**Comment by bradymiller** [comment]
hi @stephenwaite and @thesmallstar , Code looks good to me. Ready for integration. Thanks for the code improvement! -brady

**Pull request merged**